WalletWasabi/WebClients/Wasabi/HttpClientFactory.cs

```diff
@@ -91,24 +91,24 @@ public IHttpClient NewHttpClient(Func<Uri>? baseUriFn, Mode mode)
 		}
 
 		/// <summary>Creates new <see cref="TorHttpClient"/>.</summary>
-		/// <remarks>Do not use this function unless <see cref="NewHttpClient(Func{Uri}, Mode)"/> is not sufficient for your use case.</remarks>
+		/// <remarks>Do not use this function unless <see cref="NewHttpClient(Func{Uri}, Mode, ICircuit?)"/> is not sufficient for your use case.</remarks>
 		/// <exception cref="InvalidOperationException"/>
-		public TorHttpClient NewTorHttpClient(Mode mode, Func<Uri>? baseUriFn = null)
+		public TorHttpClient NewTorHttpClient(Mode mode, Func<Uri>? baseUriFn = null, ICircuit? circuit = null)
 		{
 			if (TorEndpoint is null)
 			{
 				throw new InvalidOperationException("Tor is not enabled in the user settings.");
 			}
 
-			return (TorHttpClient)NewHttpClient(baseUriFn, mode);
+			return (TorHttpClient)NewHttpClient(baseUriFn, mode, circuit);
 		}
 
 		/// <summary>
 		/// Creates a new <see cref="IHttpClient"/> with the base URI is set to Wasabi Backend.
 		/// </summary>
-		public IHttpClient NewBackendHttpClient(Mode mode)
+		public IHttpClient NewBackendHttpClient(Mode mode, ICircuit? circuit = null)
```

Comment:
Is there a way to make this a bit easier to understand the factory? Dummy example, but something like NewBackendSingleIdentitiyHttpClient.

Reply:
Yeah